metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_shares.py

```diff
@@ -0,0 +1,169 @@
+        self,
+        config: SnowflakeV2Config,
+        report: SnowflakeV2Report,
+        dataset_urn_builder: Callable[[str], str],
+    ) -> None:
+        self.config = config
+        self.report = report
+        self.logger = logger
+        self.dataset_urn_builder = dataset_urn_builder
+
+    def get_workunits(
+        self, databases: List[SnowflakeDatabase]
+    ) -> Iterable[MetadataWorkUnit]:
+        logger.debug("Checking databases for inbound or outbound shares.")
+        for db in databases:
+            inbound, outbounds = self.get_sharing_details(db)
+
+            if not (inbound or outbounds):
+                logger.debug(f"database {db.name} is not a shared.")
+                continue
+
+            sibling_dbs: List[SnowflakeDatabaseDataHubId]
+            if inbound:
+                sibling_dbs = [inbound]
+                logger.debug(f"database {db.name} is created from inbound share.")
+            else:  # outbounds
+                sibling_dbs = outbounds
+                logger.debug(f"database {db.name} is shared as outbound share.")
+
+            for schema in db.schemas:
+                for table_name in schema.tables + schema.views:
+                    # TODO: If this is outbound database,
+                    # 1. attempt listing shares using `show shares` to identify name of share associated with this database (cache query result).
+                    # 2. if corresponding share is listed, then run `show grants to share <share_name>` to identify exact tables, views included in share.
+                    # 3. emit siblings only for the objects listed above.
+                    # This will work only if the configured role has accountadmin role access OR is owner of share.
+                    # Otherwise ghost nodes may be shown in "Composed Of" section for tables/views in original database which are not granted to share.
+                    yield from self.get_siblings(
+                        db.name,
+                        schema.name,
+                        table_name,
+                        True if outbounds else False,
+                        sibling_dbs,
+                    )
+
+                    if inbound:
+                        # SnowflakeLineageExtractor is unaware of database->schema->table hierarchy
+                        # hence this lineage code is not written in SnowflakeLineageExtractor
+                        # also this is not governed by configs include_table_lineage and include_view_lineage
+                        yield self.get_upstream_lineage_with_primary_sibling(
+                            db.name, schema.name, table_name, inbound
+                        )
+
+        self.report_missing_databases(databases)
+
+    def get_sharing_details(self, db):
+        inbound = (
+            self.config.inbound_shares_map.get(db.name)
+            if self.config.inbound_shares_map
+            and db.name in self.config.inbound_shares_map
+            else None
```

Comment:
I think this can be more simply `self.config.inbound_shares_map and self.config.inbound_shares_map.get(db.name)` or `(self.config.inbound_shares_map or {}).get(db.name)`